onadata/libs/utils/csv_import.py

```diff
@@ -251,9 +251,43 @@ def submit_csv(username, xform, csv_file, overwrite=False):
     xl_datetime_columns = [
         dt.get('name') for dt in x_json.get('children')
         if dt.get('type') in XLS_DATETIME_FIELDS]
+    xl_integer_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'integer'
+    ]
+    xl_decimal_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'decimal'
+    ]
 
     try:
         for row in csv_reader:
+            for key in xl_integer_columns:
+                # For some reason Integer fields are not validated
+                # We check that the passed in data is a valid interger
+                val = row.get(key, '')
+                if val:
+                    try:
+                        interger = int(val)
+                    except ValueError:
+                        raise Exception(
+                                'Unknown interger format: {}'.format(val))
+                    else:
+                        row.update({key: interger})
+
+            for key in xl_decimal_columns:
```

Comment:
I don't know if we want to redo this now, but there's a lot of repetition here to line 311 that we could abstract with a function that takes a lambda for type conversation, I think that'd be the only argument, everything else appears to be repeated

Reply:
Very true, I'll refactor the code to be less repetitive. I think it might take more arguments something a long the lines of:

```
def _check_datatype_constraint(
        row,
        column,
        err_msg,
        constraint_check):
    for key in column:
        value = row.get(key, '')

        if value:
            try:
                value = constraint_check(value)
            except ValueError:
                raise Exception(err_msg + value)
            else:
                row.update({key: value})
```